R/sampling.R

```diff
@@ -146,3 +144,210 @@ sample_combinations <- function(ntrain, ntest, nsamples, joint_sampling = TRUE)
+#' function, which is a reimplementation of the same method. Setting this parameter to \code{"on_error"} (default)
+#' falls back to  \code{partykit::ctree}, if \code{party::ctree} fails. Other options are \code{"never"}, which always
+#' uses \code{party::ctree}, and \code{"always"}, which alsways uses \code{partykit::ctree}. A warning message is
+#' created whenever \code{partykit::ctree} is used.
+#'
+#' @return List with conditional inference tree and the variables conditioned/not conditioned on.
+#'
+#' @keywords internal
+#'
+#' @examples
+#' m <- 10
+#' n <- 40
+#' n_samples <- 50
+#' mu <- rep(1, m)
+#' cov_mat <- cov(matrix(rnorm(n * m), n, m))
+#' x_train <- data.table::data.table(MASS::mvrnorm(n, mu, cov_mat))
+#' given_ind <- c(4, 7)
+#' mincriterion <- 0.95
+#' minsplit <- 20
+#' minbucket <- 7
+#' sample <- TRUE
+#' create_ctree(given_ind = given_ind, x_train = x_train,
+#' mincriterion = mincriterion, minsplit = minsplit,
+#' minbucket = minbucket, use_partykit = "on_error")
+#'
+#' @author Annabelle Redelmeier, Martin Jullum
+#'
+#' @export
+create_ctree <- function(given_ind,
+                             x_train,
+                             mincriterion,
+                             minsplit,
+                             minbucket,
+                             use_partykit = "on_error") {
+  dependent_ind <- (1:dim(x_train)[2])[-given_ind]
+
+  if (length(given_ind) %in% c(0, ncol(x_train))) {
+    datact <- list()
+  } else {
+    y <- x_train[, dependent_ind, with = FALSE]
+    x <- x_train[, given_ind, with = FALSE]
+    df <- data.table::data.table(cbind(y, x))
+    colnames(df) <- c(paste0("Y", 1:ncol(y)), paste0("V", given_ind))
+
+    ynam <- paste0("Y", 1:ncol(y))
+    fmla <- as.formula(paste(paste(ynam, collapse = "+"), "~ ."))
+
+    # Run party:ctree if that works. If that fails, run partykit instead
+    if (use_partykit == "on_error") {
+      datact <- tryCatch(expr = {
+        party::ctree(fmla,
+                     data = df,
+                     controls = party::ctree_control(minbucket = minbucket,
+                                                     mincriterion = mincriterion))
+      }, error = function(ex) {
+        warning("party::ctree ran into the error: ", ex, "Using partykit::ctree instead!")
+        partykit::ctree(fmla,
+                        data = df,
+                        control = partykit::ctree_control(minbucket = minbucket,
+                                                          mincriterion = mincriterion,
+                                                          splitstat = "maximum"))
+      })
+    } else if (use_partykit == "never") {
+      datact <- party::ctree(fmla,
+                             data = df,
+                             controls = party::ctree_control(minbucket = minbucket,
+                                                             mincriterion = mincriterion))
+    } else {
```

Comment:
This is probably not important, but you don't check that use_partykit is "never", "always" or "on_error" so the else statement could be anything (not just for "always".

Reply:
Ok now?

Reply:
I don't see the difference?

Reply:
Select "show all commits" or "commits since last time you reviewed" in the upper left corner under the file changes view.

Reply:
@aredelmeier Let me know if you still can't see it and/or you have found it and it is ready for merge.

Reply:
Great. Now I see it.